catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -240,27 +347,36 @@ private void writeResourceToZip(
 
   private BinaryContent getBinaryContentFromZip(String filePath)
       throws CatalogTransformerException {
-    BinaryContent binaryContent;
-    InputStream fileInputStream;
+    byte[] bytes;
     File zipFile = new File(filePath);
     try {
-      fileInputStream = new ZipInputStream(new FileInputStream(zipFile));
-      binaryContent = new BinaryContentImpl(fileInputStream);
-    } catch (FileNotFoundException e) {
-      throw new CatalogTransformerException("Unable to get ZIP file from ZipInputStream.", e);
+      bytes = Files.readAllBytes(zipFile.toPath().toAbsolutePath());
+    } catch (IOException e) {
+      throw new CatalogTransformerException(
+          "An error occurred while streaming the temporary zip file", e);
     }
 
-    jarSigner.signJar(
-        zipFile,
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty(SystemBaseUrl.EXTERNAL_HOST)),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStorePassword")),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStore")),
-        AccessController.doPrivileged(
-            (PrivilegedAction<String>) () -> System.getProperty("javax.net.ssl.keyStorePassword")));
-    return binaryContent;
+    AccessController.doPrivileged(
+        (PrivilegedAction<Void>)
+            () -> {
+              jarSigner.signJar(
+                  zipFile,
+                  AccessController.doPrivileged(
```

Comment:
✏️ ❓ You may not need these nested `doPrivileged` calls, since all of this is within a `doPrivileged` call.